crates/bevy_ecs/src/spawn.rs

```diff
@@ -0,0 +1,358 @@
+pub trait SpawnRelated: RelationshipTarget {
+    /// Returns a [`Bundle`] containing this [`RelationshipTarget`] component. It also spawns a [`SpawnableList`] of entities, each related to the bundle's entity
+    /// via [`RelationshipTarget::Relationship`]. The [`RelationshipTarget`] (when possible) will pre-allocate space for the related entities.
+    ///
+    /// See [`Spawn`], [`SpawnIter`], and [`SpawnWith`] for usage examples.
+    fn spawn<L: SpawnableList<Self::Relationship>>(
+        list: L,
+    ) -> SpawnRelatedBundle<Self::Relationship, L>;
+
+    /// Returns a [`Bundle`] containing this [`RelationshipTarget`] component. It also spawns a single entity containing [`Bundle`] that is related to the bundle's entity
+    /// via [`RelationshipTarget::Relationship`].
+    ///
+    /// ```
+    /// # use bevy_ecs::hierarchy::Children;
+    /// # use bevy_ecs::spawn::SpawnRelated;
+    /// # use bevy_ecs::name::Name;
+    /// # use bevy_ecs::world::World;
+    /// let mut world = World::new();
+    /// world.spawn((
+    ///     Name::new("Root"),
+    ///     Children::spawn_one(Name::new("Child")),
+    /// ));
+    /// ```
+    fn spawn_one<B: Bundle>(bundle: B) -> SpawnOneRelated<Self::Relationship, B>;
+}
+
+impl<T: RelationshipTarget> SpawnRelated for T {
+    fn spawn<L: SpawnableList<Self::Relationship>>(
+        list: L,
+    ) -> SpawnRelatedBundle<Self::Relationship, L> {
+        SpawnRelatedBundle {
+            list,
+            marker: PhantomData,
+        }
+    }
+
+    fn spawn_one<B: Bundle>(bundle: B) -> SpawnOneRelated<Self::Relationship, B> {
+        SpawnOneRelated {
+            bundle,
+            marker: PhantomData,
+        }
+    }
+}
+
+/// Returns a [`SpawnRelatedBundle`] that will insert the given [`RelationshipTarget`], spawn a [`SpawnableList`] of entities with given bundles that
+/// relate to the [`RelationshipTarget`] entity via the [`RelationshipTarget::Relationship`] component, and reserve space in the [`RelationshipTarget`] for each spawned entity.
+///
+/// The first argument is the [`RelationshipTarget`] type. Any additional arguments will be interpreted as bundles to be spawned.
+///
+/// Also see [`children`](crate::children) for a [`Children`](crate::hierarchy::Children)-specific equivalent.
+///
+/// ```
+/// # use bevy_ecs::hierarchy::Children;
+/// # use bevy_ecs::name::Name;
+/// # use bevy_ecs::world::World;
+/// # use bevy_ecs::related;
+/// # use bevy_ecs::spawn::{Spawn, SpawnRelated};
+/// let mut world = World::new();
+/// world.spawn((
+///     Name::new("Root"),
+///     related!(Children[
+///         Name::new("Child1"),
+///         (
+///             Name::new("Child2"),
+///             related!(Children[
+///                 Name::new("Grandchild"),
+///             ])
+///         )
+///     ])
+/// ));
+/// ```
+#[macro_export]
+macro_rules! related {
+    ($relationship_target:ident [$($child:expr),*$(,)?]) => {
+       $relationship_target::spawn(($(Spawn($child)),*))
```

Comment:
it would be nice to avoid having to explicitly import `Spawn` for this, even if it's in the prelude
```suggestion
       $relationship_target::spawn(($($crate::spawn::Spawn($child)),*))
```

Reply:
didn't see MrGVSV's comment above, oops 🙃